vendor/github.com/olivere/elastic/client.go

```diff
@@ -381,7 +381,7 @@ func SetHealthcheckInterval(interval time.Duration) ClientOptionFunc {
 
 // SetMaxRetries sets the maximum number of retries before giving up when
 // performing a HTTP request to Elasticsearch.
-func SetMaxRetries(maxRetries int) func(*Client) error {
+func SetMaxRetries(maxRetries int) ClientOptionFunc {
```

Comment:
Yeah I totally agree. But they've already fixed this in the [`release-branch.v3`](https://github.com/olivere/elastic/blob/release-branch.v3/client.go#L479), so it will be available soon. All I've done is implement a change that's required to get the types right, so no risk of vendor drift.